### DISCUSSION TITLE
How to make a loop?

### DISCUSSION THREAD

**Discussion by Inferdy** [mxgmn/MarkovJunior#31] [Q&A] [answered]
I want to make sequence repeat
- an infinite
- a finite

number of times, it has some nodes with steps attributes. How can I do this?

**Comment by mxgmn** [comment, marked as answer]
To make a sequence repeat, you need to put a `sequence` node inside a `markov` node. Examples of this: [HamiltonianPath](https://github.com/mxgmn/MarkovJunior/blob/main/models/HamiltonianPath.xml), [SelectLargeCaves](https://github.com/mxgmn/MarkovJunior/blob/main/models/SelectLargeCaves.xml), [SelectLongKnots](https://github.com/mxgmn/MarkovJunior/blob/main/models/SelectLongKnots.xml), [FireNoise](https://github.com/mxgmn/MarkovJunior/blob/main/models/FireNoise.xml), [SmartSAW](https://github.com/mxgmn/MarkovJunior/blob/main/models/SmartSAW.xml), [FindLongCycle](https://github.com/mxgmn/MarkovJunior/blob/main/models/FindLongCycle.xml). Counters in markov/sequence nodes are not supported right now. Instead, you may want to repeat the sequence until some node is matched.

**Reply by Inferdy** [reply]
To be clear we must say that counters do work inside the cycles, but they cannot be attached to the cycles themselves.
Here's an example of the counters (steps) inside a cycle:

![6jwpx7](https://user-images.githubusercontent.com/24939153/174145414-00105723-c92e-4c29-9578-23fee2cebde0.gif)
```xml
<sequence values="BRWGO" origin="Center">
   <markov>
      <sequence>
         <all in="BBB/BRB/BBB" out="WWW/W*W/WWW" steps="1"/>
         <one in="R***B" out="****G" steps="3"/>
         <prl in="R***G" out="**W**" steps="1"/>
         <prl in="R" out="W"/>
         <prl in="G" out="R"/>
      </sequence>
   </markov>
   <markov>
      <sequence>
         <one in="BBWBB/BWWWB/BWWWB/BWWWB/BBBBB" out="*****/*****/**R**/*****/*****" steps="1"/>
         <one in="R*B*W" out="G*W**" steps="1"/>
         <prl in="G*B*W" out="**W**" steps="1" p="0.35"/>
         <one in="G" out="W" steps="1"/>
         <one in="R" out="O" steps="1"/>
      </sequence>
   </markov>
</sequence>
```

**Reply by mxgmn** [reply]
> counters do work inside the cycles, but they cannot be attached to the cycles themselves

Yes, exactly.
